src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -831,5 +832,46 @@ private void EmbedSourceLink(Stream stream)
                 kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.SourceLink),
                 value: _debugMetadataOpt.GetOrAddBlob(bytes));
         }
+
+        private void EmbedCompilerFlags(CommonPEModuleBuilder module)
+        {
+            byte[] bytes = new byte[0];
+
+            // TODO: Where to get compiler version? 
+
+            // TODO: Where to get source encoding
+
+            _debugMetadataOpt.AddCustomDebugInformation(
+                parent: EntityHandle.ModuleDefinition,
+                kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.DeterministicCompilerFlags),
+                value: _debugMetadataOpt.GetOrAddBlob(bytes));
+        }
+
+        private void EmbedMetadataReferenceInformation(CommonPEModuleBuilder module)
+        {
+            var bytes = new List<byte>();
+
+            // TODO: Get metadata reference information
+            // File name: Foo.exe
+            // COFF header Timestamp field (4 bytes): 0x542d5742
+            // COFF header SizeOfImage field (4 bytes): 0x32000000
+            // MVID (Guid, 24 bytes): 0x24a44d8218894463807674caf3b1c19a
+            // Output: Foo.exe\0542d5742320000000024a44d8218894463807674caf3b1c19a
+            foreach (var metadataReference in module.CommonCompilation.ExternalReferences)
+            {
+                if (metadataReference.Display is object && !IsTooLongInternal(metadataReference.Display, PdbLengthLimit))
+                {
+                    bytes.AddRange(Encoding.UTF8.GetBytes(metadataReference.Display));
+                    bytes.Add(0);
+
+                    // TODO: How to get the timestamp, sizeof, and mvid?
```

Comment:
>// TODO: How to get the timestamp, sizeof, and mvid? [](start = 20, length = 52)

You'll need to get `PEAssembly` object for the reference and from there `ManifestModule` gives you `PEModule` and that has `PEReaderOpt` property. If it's null we can skip such reference as it was given as just metadata image without PE headers. This scenario won't happen when building from command line or msbuild, so we can ignore that.

Reply:
I couldn't quite figure out where the PEAssembly was, so I hacked this (very obviously wrong) thing together to make sure I was using the reader correctly.